### DISCUSSION TITLE
modbus esp32

### DISCUSSION THREAD

**Discussion by freedreamer82** [RavenSystem/esp-homekit-devices#2045] [New Features]
are you @RavenSystem  going to add  in the future a modbus (rs485-rs232) support ? should be interesting since a lot of devices are based on this old style protocol.

Marco

**Comment by RavenSystem** [comment]
https://github.com/RavenSystem/esp-homekit-devices/wiki/General-Configuration#mode-only-esp32-models

**Comment by freedreamer82** [comment]
yes I already saw the phy , I was wondering also the stack modbus , to query and parse the registers to transform into homekit values.

**Reply by RavenSystem** [reply]
This is already done by UART actions (Send queries) and UART Receiver from Free Monitor Service (Receive and parse responses).

**Reply by freedreamer82** [reply]
I think I did not understand... I checked the free monitor and it useful to monitor the rs485 but i cannot understand how to perform a Read Holding Registers modbus. Do you any example for read/write ? and what about CRC , who will add it ?

**Reply by PhaseAngle** [reply]
Looks like it supports rs485 but not modbus protocol.
Too many people don't know there is a difference.
I mostly use modbus over ethernet on my solar equipment.

**Reply by RavenSystem** [reply]
RS485 is a communication bus, like UART or I2C. Other thing is what you send or receive over these buses. HAA works at a lower layer, and user must implement upper protocol layers by himself into MEPLHAA scripts. You must know how technically ModBus works over RS485, and send property bytes over RS485 bus using UART Actions, and receive and process response bytes using UART receiver of Free Monitor.

**Reply by oxbown** [reply]
@freedreamer82 @RavenSystem any examples with UART receaver in Free Monitor mode?? I´m trying to implement modbus accesories.
